### PR TITLE
docs(storybook): interaction tests updated documentation

### PR DESCRIPTION
**Pull request by mandarini**
Documentation for Interaction Tests

### CODE REVIEW

**Review by jaysoo — APPROVED** [review]

File: docs/generated/packages/storybook/documents/storybook-interaction-tests.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
+---
+title: Setting up Storybook interaction tests with Nx
+description: This guide explains how you can set up Storybook interaction tests on your Nx workspace.
+---
+
+# Setting up Storybook interaction tests with Nx
```
Comment (juristr) [inline docs/generated/packages/storybook/documents/storybook-interaction-tests.md:6]: We should use the title form with upper case words, forgot the proper name, but basically

"Setting up Storybook Interaction Tests with Nx"

File: docs/generated/packages/storybook/documents/storybook-interaction-tests.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+## The `play` Function
+
+The [play function](https://storybook.js.org/docs/react/writing-stories/play-function) is a mechanism provided by Storybook to simulate user interactions within a story. By adding the play function to your stories, you can programmatically interact with the rendered elements, simulating clicks, typing, and more. This function serves as a bridge between your Storybook UI and your testing logic, making it easy to write interaction tests. Since you already have stories configured, it's just one small step to add interaction tests to your Storybook setup.
+
+## The test runner
+
+[Storybook's test-runner](https://storybook.js.org/docs/react/writing-tests/test-runner) is a tool that lets you run your interaction tests, powered by Jest and Playwright. It runs parallel to your Storybook. It works seamlessly with the play function, allowing you to execute the interactions you've defined in your stories and assert their results, ensuring your components behave correctly.
```
Comment (juristr) [inline docs/generated/packages/storybook/documents/storybook-interaction-tests.md:20]: I'm wondering whether we actually need all this. It feels like this should be on the storybook docs. We already link to those docs in the intro sentence, so I wouldn't add all this info. The aim is to keep things as short as possible

File: docs/generated/packages/storybook/documents/storybook-interaction-tests.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+## How to Set Up Interaction Testing in Nx
```
Comment (juristr) [inline docs/generated/packages/storybook/documents/storybook-interaction-tests.md:22]: We don't need this title. It is the same as the one of the entire document here.

File: docs/generated/packages/storybook/documents/storybook-interaction-tests.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+{% callout type="warning" title="Set up Storybook in your workspace" %}
+You first need to set up Storybook for your Nx workspace, if you haven't already. You can read the [Storybook plugin overview guide](/packages/storybook) to get started.
+{% /callout %}
+
+### Set Up Storybook Interaction Tests
```
Comment (juristr) [inline docs/generated/packages/storybook/documents/storybook-interaction-tests.md:28]: Make this an H2 title and rename it to "Setup"

File: docs/generated/packages/storybook/documents/storybook-interaction-tests.md [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+{% /tabs %}
+
+This command will:
+
+- [Set up Storybook for your project](/packages/storybook) - including the `@storybook/addon-interactions` addon.
+- Add a `play` function to your stories.
+- Install the necessary dependencies.
+- Generate a `test-storybook` target in your project's `project.json`, which has a command to invoke the Storybook test runner.
+
+### Writing an Interaction Test
```
Comment (juristr) [inline docs/generated/packages/storybook/documents/storybook-interaction-tests.md:65]: Make it a H2 title

File: docs/generated/packages/storybook/documents/storybook-interaction-tests.md [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+    );
+  },
+};
+```
+
+Here, the play function is simulating a click on the button using the `userEvent.click` method. It is then asserting that the button's text has changed to reflect the number of times it has been clicked.
+
+You can read more about how to write interaction tests in the [Storybook documentation](https://storybook.js.org/docs/react/writing-tests/interaction-testing).
+
+### Running Interaction Tests
```
Comment (juristr) [inline docs/generated/packages/storybook/documents/storybook-interaction-tests.md:115]: H2 title

File: docs/generated/packages/storybook/documents/storybook-interaction-tests.md [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+
+Setting up interaction tests with Nx and Storybook provides an extra layer of confidence in the functionality of your components. It ensures that they not only look right but also behave correctly in response to user interactions.
+
+Storybook interaction tests provide a unique advantage over traditional e2e tests, especially when considering the development setup. With Storybook already in place, you essentially have a controlled environment set up for each of your components. This allows you to write interaction tests almost immediately, without the overhead of setting up and navigating through a full application environment, as is the case with e2e tests.
+
+Moreover, since Storybook isolates each component, you can ensure that the tests are solely focused on individual component behavior rather than application-level concerns. This results in faster test execution, easier debugging, and more granular feedback during the development process. In essence, with Storybook's interaction tests, you get many of the benefits of e2e tests but with a setup that's quicker, more focused, and integrated right into your component development workflow.
```
Comment (juristr) [inline docs/generated/packages/storybook/documents/storybook-interaction-tests.md:135]: Again, I feel like this is a section for the storybook docs. The section is still useful though, but as an Nx user my main question would be why I would want to use this approach now vs the approach Nx has been promoting in the past, namely to use Cypress e2e tests for Storybook stories.

File: docs/shared/packages/storybook/interaction-tests.md [hunk 8]
```diff
@@ -0,0 +1,142 @@
+---
+title: Setting up Storybook interaction tests with Nx
+description: This guide explains how you can set up Storybook interaction tests on your Nx workspace.
+---
+
+# Setting up Storybook interaction tests with Nx
```
Comment (juristr) [inline docs/shared/packages/storybook/interaction-tests.md:6]: Just noticed I commented on the generated MD file, so plz have a look at my comments above 😅

**Review by juristr — CHANGES_REQUESTED** [review]

File: docs/generated/packages/storybook/documents/storybook-interaction-tests.md (hunk 7 above)
Comment (mandarini) [reply docs/generated/packages/storybook/documents/storybook-interaction-tests.md:135]: Cool I'll change that section to explain our focus switch!

**Review by juristr — APPROVED** [review]

**Pull request merged**